src/client/api.ts

```diff
@@ -83,11 +103,18 @@ export function buildApi(
             container.registerProvider(picker);
         },
         onKernelStateChange: notebookExtensibility.onKernelStateChange.bind(notebookExtensibility),
-        registerCellToolbarButton: webviewExtensibility.registerCellToolbarButton.bind(webviewExtensibility)
+        registerCellToolbarButton: webviewExtensibility.registerCellToolbarButton.bind(webviewExtensibility),
+        registerNewNotebookContent(options: { defaultCellLanguage: string; label?: string }) {
+            return serviceContainer.get<CreationOptionService>(CreationOptionService).registerNewNotebookContent(options);
+        },
+        createBlankNotebook: async (options: { defaultCellLanguage: string }): Promise<void> => {
+            const service = serviceContainer.get<INotebookEditorProvider>(VSCodeNotebookProvider);
+            await service.createNew(options);
+        }
     };
 
     // In test environment return the DI Container.
-    if (isTestExecution()) {
+    if (isTestExecution() || process.env.VSC_JUPYTER_EXPOSE_SVC) {
```

Comment:
Jupyter extension activates early due to the Python extension loading early even before the test on my machine, setup an env variable that's specific for this to get around that.